### ISSUE TITLE
[Bug]: libfcl build failed

### ISSUE THREAD

**Issue by johndebia2022** [termux/termux-packages#12219]
-- Performing Test _compiler_supports_sse - Failed
CMake Error at /data/data/com.termux/files/usr/share/octomap/octomap-targets.cmake:105 (message):
  The imported target "octomath-static" references the file

     "/data/data/com.termux/files/usr/lib/liboctomath.a"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/data/data/com.termux/files/usr/share/octomap/octomap-targets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /data/data/com.termux/files/usr/share/octomap/octomap-config.cmake:77 (include)
  CMakeLists.txt:241 (find_package)

-- Configuring incomplete, errors occurred!
See also "/home/builder/.termux-build/libfcl/build/CMakeFiles/CMakeOutput.log".
See also "/home/builder/.termux-build/libfcl/build/CMakeFiles/CMakeError.log".
builder@5b3c90e51e03:~/termux-packages$

[libfcl.txt](https://github.com/termux/termux-packages/files/9690278/libfcl.txt)

**Comment by pvonmoradi** [comment]
Title is wrong? (issue is unrelated to `libfcl`)

**Comment by johndebia2022** [comment]
> Title is wrong? (issue is unrelated to `libfcl`)

Updated

**Comment by pvonmoradi** [comment]
Yes, can confirm. It seems there is something wrong with the way fcl uses octomap. It shouldn't be calling its static version.
If we build octomap itself, static libraries are built:
```
./scripts/run-docker.sh ls -alhF /data/data/com.termux/files/usr/lib
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
total 1.4M
drwxr-xr-x 3 builder builder 4.0K Oct  1 13:13 ./
drwxr-xr-x 9 builder builder 4.0K Oct  1 13:13 ../
-rw-r--r-- 1 builder builder  35K Oct  1 13:13 libdynamicedt3d.a
-rw-r--r-- 1 builder builder  66K Oct  1 13:13 libdynamicedt3d.so
-rw-r--r-- 1 builder builder 767K Oct  1 13:13 liboctomap.a
-rw-r--r-- 1 builder builder 415K Oct  1 13:13 liboctomap.so
-rw-r--r-- 1 builder builder  44K Oct  1 13:13 liboctomath.a
-rw-r--r-- 1 builder builder  66K Oct  1 13:13 liboctomath.so
-rw-r--r-- 1 builder builder   11 Oct  1 13:13 libutil.so
drwxr-xr-x 2 builder builder 4.0K Oct  1 13:13 pkgconfig/
```
But if we build libfcl, only `.so` version octomap/octomath would be built.

I can disable optional octomap dependency of libfcl for the time being if that is desired....

**Comment by truboxl** [comment]
Closing since the build issue was fixed in #12232 unless octomap want to use `TERMUX_PKG_NO_STATICSPLIT=true`
